feat(gateway): add multi-target gateway mode (#21)
rebellions-sw/rbln-metrics-exporter: README.md

      - targets:
          - npu-host-1:50051   # rbln-smd gRPC address on each NPU host
          - npu-host-2:50051
    relabel_configs:
      # 1. Copy the listed address into the ?target= query parameter.
      - source_labels: [__address__]
        target_label: __param_target
      # 2. Keep it as the instance label so series stay distinguishable per host.
      - source_labels: [__param_target]
        target_label: instance
      # 3. Send the actual HTTP request to the gateway instead of the NPU host.
      - target_label: __address__
        replacement: rbln-gateway:9200   # gateway exporter address
```

With this in place Prometheus scrapes `http://rbln-gateway:9200/metrics?target=npu-host-1:50051` for each host and stores every series with `instance="npu-host-1:50051"`. Rule order matters: the copy rules must run before `__address__` is overwritten.

### Alerting

Two failure signals exist and mean different things:

| Expression | Meaning |
| --- | --- |
| `up == 0` | The gateway itself (or the network path to it) is down — affects all targets |
| `up == 1 and rbln_up == 0` | That specific NPU host's `rbln-smd` is unreachable |

### Notes and Limitations

> - Collection happens inside the scrape request, so slow targets consume scrape time. The gateway honors the `X-Prometheus-Scrape-Timeout-Seconds` header Prometheus sends (10s cap by default) — raise `scrape_timeout` for slow links.

Reviewer: "cap" is inaccurate for the current behavior — when the header is present its value is used as-is (uncapped); 10s is only the fallback when the header is absent:

```suggestion
- Collection happens inside the scrape request, so slow targets consume scrape time. The gateway honors the `X-Prometheus-Scrape-Timeout-Seconds` header Prometheus sends (falling back to 10s when the header is absent) — raise `scrape_timeout` for slow links.
```

(If a `--scrape-timeout` cap flag lands, this becomes a real cap — reword accordingly.)
